components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -269,7 +269,9 @@ void ConversationDriver::InitEngine() {
   auto* model = GetModel(model_key_);
   // Make sure we get a valid model, defaulting to static default or first.
   if (!model) {
-    NOTREACHED() << "Model was not part of static model list";
+    // It is unexpected that we get here. Dump a call stack
+    // to help figure out why it happens.
+    base::debug::DumpWithoutCrashing();
```

Comment:
can we report the faulty `model_key_` here? 

this is such an awkward and i can't reproduce it.